### PR TITLE
Form: Add alpha support for patternProperties

### PR DESCRIPTION
**Pull request by LucianBuzzo**
This change adds support to the Form component for the JSON schema
"patternProperties" keyword. This is an early implementation and needs
further work to be production ready.
I also took this opportunity to tidy up JSDOM setup, by defining the
document global and other useful values in one place.

Change-type: minor
Signed-off-by: Lucian <lucian.buzzo@gmail.com>

### CODE REVIEW

File: __tests__/components/Form.spec.js [hunk 1]
```diff
@@ -405,4 +405,31 @@ describe('Form component', () => {
       component.unmount()
     })
   })
+
+  describe('patternProperties', () => {
```
Comment (cyplo) [inline __tests__/components/Form.spec.js]: are we concerned with removal of properties in this PR ?

File: __tests__/components/Form.spec.js [hunk 2]
```diff
@@ -405,4 +405,31 @@ describe('Form component', () => {
...
+
+      const component = mount(
+        <Provider>
+          <Form
+            schema={schema}
+          />
+        </Provider>
+      )
+
+      const valueInput = component.find('input.rendition-form-pattern-properties__value-field')
```
Comment (cyplo) [inline __tests__/components/Form.spec.js]: would having an additional test, that the object produced from the form is in the right shape, help ?

File: __tests__/components/Form.spec.js (hunk 1 above)
Comment (LucianBuzzo) [reply __tests__/components/Form.spec.js]: Could you expand on your question? Its a little ambiguous

File: __tests__/components/Form.spec.js (hunk 2 above)
Comment (LucianBuzzo) [reply __tests__/components/Form.spec.js]: I spent some time wrestling with Enzyme trying to get tests to run correctly, but kept encountering a very strange bug. In the end I decided to continue with the PR and come back to unit tests as I didn't want to block and alpha release of this functionality.

Comment (cyplo) [reply __tests__/components/Form.spec.js]: alrighty then :)

File: __tests__/components/Form.spec.js (hunk 1 above)
Comment (cyplo) [reply __tests__/components/Form.spec.js]: looking at the test title `should allow you to add new key value pairs` - would an opposite need to be tested ? e.g. a removal of previously added property key and value by the user of the UI ?

Comment (LucianBuzzo) [reply __tests__/components/Form.spec.js]: @cyplo The functionality for removing key/value pairs exists and works, but isn't tested yet

**Pull request merged**